R/sitevisitpeak-data.R

```diff
@@ -133,9 +133,20 @@ getIvDifference <- function(readingVal, ivVal) {
       result <- as.character(round(val, digits = nchar(ivVal)))
       
       if(abs(val) > 0.05) {
-        result <- paste(result, "*")
+        result <- paste(result, "**")
       }
     }
   }
   return(result)
 }
+
+containsOutsideUncertainty <- function(data) {
+  diff <- list()
+  data <- data$readings
+  for(listRows in row.names(data)){
+    listElements <- data[listRows,]
+    diff <- append(diff, getIvDifference(listElements$value, listElements$associatedIvValue))
+  }
+
+  return(length(diff[grepl("\\*\\*", diff)]) > 0)
+}
```

Comment:
I'm really not sure if this is the most efficient way of figuring out if any of the values are outside of the specified range before actually rendering the table, so I'm certainly open to suggestions and feedback here.